packages/reporting-server/parsers/task_summary_parser.py

```diff
@@ -0,0 +1,24 @@
+import json
+
+from models.task_summary import TaskSummary
+
+
+async def task_summary_parser(fullstring: str):
+    modified_string = fullstring.replace("task_summary:", "")
+    state_json = json.loads(modified_string)
+
+    if len(state_json) > 0:
+        return {
+            "payload": modified_string,
+            "fleet_name": state_json["fleet_name"],
+            "task_id": state_json["task_id"],
+            "task_profile": state_json["task_profile"],
+            "state": TaskSummary.service.get_task_state_name(state_json["state"]),
+            "status": state_json["status"],
+            "submission_time": state_json["submission_time"],
+            "start_time": state_json["start_time"],
+            "end_time": state_json["end_time"],
+            "robot_name": state_json["robot_name"],
+        }
```

Comment:
Just to check, are these field always going to have values? or should add nullable fields to the model? or should we add default values?

Reply:
Status is an empty field ("") sometimes, if I am not wrong. Robot name as well, when the task has not yet been assigned to the robot. I think we could include an empty "" field?